docs/source/howto/workflows.rst

```diff
@@ -4,14 +4,217 @@
 How to run multi-step workflows
 *******************************
 
-
 .. _how-to:workflows:write:
 
 Writing workflows
 =================
 
-`#3991`_
+A workflow in AiiDA is a :ref:`process <topics:processes:concepts>` that calls other workflows and calculations and optionally *returns* data and as such can encode the logic of a typical scientific workflow.
+Currently, there are two ways of implementing a workflow process:
+
+ * :ref:`work functions<topics:workflows:concepts:workfunctions>`
+ * :ref:`work chains<topics:workflows:concepts:workchains>`
+
+Here we present a brief introduction on how to write both workflow types.
+
+.. note::
+
+    For more details on the concept of a workflow, and the different between a work function and a work chain, please see the corresponding :ref:`topics section<topics:workflows:concepts>`.
+
+Work function
+-------------
+
+A *work function* is a process function that calls one or more calculation functions and *returns* data that has been *created* by the calculation functions it has called.
```

Comment:
```suggestion
A *work function* is a process function that calls one or more other process functions and *returns* data that has been *created* by the calculation functions it has called.
```
Not sure if process functions will already be clear to the reader here, but it is important to note, I think, that a work function can also call other work functions, and not just calc functions.

Now that I read it back. Maybe keep the sentence as is and add another one that says that it can also call other work functions, allowing to write nested workflows